provisioning/resources/control-plane/main.go

```diff
@@ -0,0 +1,284 @@
+/**
+ * Copyright (c) 2016-2017 Snowplow Analytics Ltd.
+ * All rights reserved.
+ *
+ * This program is licensed to you under the Apache License Version 2.0,
+ * and you may not use this file except in compliance with the Apache
+ * License Version 2.0.
+ * You may obtain a copy of the Apache License Version 2.0 at
+ * http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the Apache License Version 2.0 is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied.
+ *
+ * See the Apache License Version 2.0 for the specific language
+ * governing permissions and limitations there under.
+ */
+
+package main
+
+import (
+  "io"
+  "net/http"
+  "log"
+  "os"
+  "flag"
+  "io/ioutil"
+  "github.com/BurntSushi/toml"
+)
+
+var configPath string
+var config ControlPlaneConfig
+
+func main() {
+  configFlag := flag.String("config", "", "Control Plane API config file")
+  flag.Parse()
+  configPath = *configFlag
+
+  if _, err := toml.DecodeFile(configPath, &config); err != nil {
+    panic(err)
+  }
+
+  http.HandleFunc("/restart-services", restartServices)
+  http.HandleFunc("/enrichments", uploadEnrichments)
+  http.HandleFunc("/external-iglu", addExternalIgluServer)
+  http.HandleFunc("/local-iglu-apikey", addLocalIgluApikey)
+  http.HandleFunc("/credentials", changeUsernameAndPassword)
+  http.HandleFunc("/domain-name", addDomainName)
+  http.HandleFunc("/version", getSpminiVersion)
+  log.Fatal(http.ListenAndServe(":10000", nil))
+}
+
+func restartServices(resp http.ResponseWriter, req *http.Request) {
+  if (req.Method == "PUT") {
+    err := restartSPServices()
+    if err != nil {
+      http.Error(resp, err.Error(), 500)
+      return
+    } else {
+      resp.WriteHeader(http.StatusOK)
+      io.WriteString(resp, "OK")
+    }
+  }
+}
+
+func uploadEnrichments(resp http.ResponseWriter, req *http.Request) {
+  if req.Method == "POST" {
+    req.ParseMultipartForm(32 << 20)
```

Comment:
please explain with a comment and a variable why this size was chosen

Reply:
also why is the error discarded? same for `req.ParseForm` below

Reply:
I have nothing against the value, but I don't want to have to checkout go documentation when I stumble on this one year from now, so pls explain what those 32mb refer to in a comment and introduce a variable :+1: